paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -95,17 +99,24 @@ class PaparazziPlugin : Plugin<Project> {
         "preparePaparazzi${variantSlug}Resources",
         PrepareResourcesTask::class.java
       ) { task ->
+        fun DirectoryProperty.asRelativePathString(childDirectory: Provider<Directory>): Provider<String> =
+          flatMap { buildDir ->
+            childDirectory.map { childDir ->
+              buildDir.asFile.toPath().relativize(childDir.asFile.toPath()).invariantSeparatorsPathString
+            }
+          }
+
```

Comment:
Let's keep this but also:
-  copy over the previous relativize function
- move both out of local scope and to the bottom of the plugin class declaration

i.e., 
```
  ...

  private fun Directory.relativize(child: Directory): String {
    return asFile.toPath().relativize(child.asFile.toPath()).invariantSeparatorsPathString
  }

  private fun DirectoryProperty.asRelativePathString(child: Provider<Directory>): Provider<String> =
    flatMap { root -> child.map { root.relativize(it) } }
}

// TODO: Migrate to ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE when Gradle 7.3 is
```